File: Hivemind.Ng/src/autogenerated/entities/Injury.ts

export class Injury {
    public injuryEnum: string;
    public name: string;
    public description: string;
    public injuryEffect: any;

    public constructor(partial: Partial<Injury>) {
        if (partial.injuryEnum) {
            this.injuryEnum = partial.injuryEnum;
        }
        if (partial.name) {
            this.name = partial.name;
        }
        if (partial.description) {
            this.description = partial.description;
        }
        if (partial.injuryEffect) {
            this.injuryEffect = partial.injuryEffect;
        }
    }
}

File: Hivemind.Core/Entities/Injury.cs
// <copyright file="Injury.cs" company="weirdvector">
// Copyright (c) weirdvector. All rights reserved.
// </copyright>

using System;
using Hivemind.Enums;

namespace Hivemind.Entities
{
    /// <summary>
    /// Represents an Injury
    /// </summary>
    public class Injury
    {
        /// <summary>
        /// Gets or sets the InjuryId,
        /// </summary>
        public InjuryEnum InjuryId { get; set; }

        /// <summary>
        /// Gets or sets the Name.
        /// </summary>
        public string Name { get; set; }

        /// <summary>
        /// Gets or sets the Description
        /// </summary>
        public string Description { get; set; }

        /// <summary>
        /// Gets or sets the function representing the injury's effect on a ganger.
        /// </summary>
        public Func<Ganger, Ganger> InjuryEffect { get; set; }
    }
}
